src/stitch_fix/logWeasel.js

```diff
@@ -0,0 +1,19 @@
+import { ulid } from 'ulid'
+
+class LogWeasel {
+
+  constructor(key) {
+    if (typeof key === 'undefined') {
+      throw new TypeError('key must be supplied')
+    }
+
+    this.key = key;
+  }
+
+  generateId () {
+    const formattedKey = this.key + '-WEB';
+    return ulid() + '-' + formattedKey;
+  }
+}
```

Comment:
What would you think about writing this as a function vs a class? It looks like we're taking a value in, transforming it, and pushing a value out. That could be a bit simpler (as in less code when compiled, less to consider) as a simple key in, content out function.

At a glance, it looks like it could be as simple as

```js
const logWeaselId = (key) => {
  // check for key and throw
  return `${ulid()}-${key}-WEB`;
}
```

Reply:
So yeah, this is what I went with at first, and can definitely go back to. I decided to use the class because I liked how it would read when the class was consumed:
```
logWeasel = new LogWeasel("CLIENT_HOME_UI");
logWeasel.generateId();
```

Also, in a SPA world, you'd only need to instantiate `LogWeasel` once, and then for each async call, just call `generateId()`. At least that's what I had in my head. Thoughts?

Reply:
I think most JS developers are more comfortable working with simple functions, so maintenance over time would be  more welcoming without the extra considerations that come with JS classes. Since instantiation doesn't memoize any  heavy operations, I think the function would be more performant as well. 

Usage for the function I imagine would look something like:
```js
const logWeaselID = generateLogWeaselD(key);
```
which for me at least is very pleasant.

Reply:
Sounds good to me. Will make that change.